modules/file-rotation/file-rotation.c

```diff
@@ -22,29 +22,47 @@
  */
 
 #include "file-rotation.h"
+#include "modules/http/http-signals.h"
+#include "modules/affile/file-signals.h"
 #include "driver.h"
 
-struct _FileRotationPlugin
+static void
+_slot_append_file_rotation_request(FileRotationPlugin *self, FileFlushSignalData *data)
 {
-  LogDriverPlugin super;
-  gsize size;
-};
+
+  g_string_append_printf(self->filename, "%s", data->filename->str);
+}
 
 static gboolean
-file_rotation_attach_to_driver(LogDriverPlugin *s, LogDriver *driver)
+file_rotation_attach_to_driver(LogDriverPlugin *s, FileRotationPlugin *driver)
```

Comment:
No need to change the signature of this function.

You can get a signal-slot-connector instance this way:

```
SignalSlotConnector *ssc = driver->super.signal_slot_connector;
```

Reply:
Changed :+1: